azure/scope/managedcontrolplane.go

```diff
@@ -198,6 +198,14 @@ func (s *ManagedControlPlaneScope) NodeSubnet() infrav1.SubnetSpec {
 	}
 }
 
+// SetSubnet sets the subnet spec for the subnet with the same role.
+func (s *ManagedControlPlaneScope) SetSubnet(subnetSpec infrav1.SubnetSpec) {
+	s.ControlPlane.Spec.VirtualNetwork.Subnet = infrav1exp.ManagedControlPlaneSubnet{
+		Name:      subnetSpec.Name,
+		CIDRBlock: subnetSpec.CIDRBlocks[0],
+	}
+}
```

Comment:
should this be a no op?

Reply:
Since the function is not used when using a managed control plane, I think you are right.